Weather app using APIs

bertil291utn opened the pull request:
# Weather app

### Please I'll be grateful if you send me some feedback about my project.

We use 
- ES6 Modules
- Factory Functions
- Local Storage
- UI/UX design 
- Webpack



This is the [project's repository](https://github.com/bertil291utn/weather-app/tree/feature/weather-app-branch), or directly the [live demo](http://weather.bertiltandayamo.me/).

Thanks in advance 💻 🚀 🔥

bertil291utn committed:
Add layout to gitignore

bertil291utn committed:
Add layout to gitignore

bertil291utn committed:
Remove layoout from this branch

bertil291utn committed:
Add dist to gitignore

bertil291utn committed:
Add sass load to webpack
- Add sass file to layout directory
- Add script to package json to run node-sass
- Add css ansd watch scss build

bertil291utn committed:
Creeate new indes html file
- Link all css files
- Add GA directory

bertil291utn committed:
Add new font family to layer

bertil291utn committed:
Add hero section according to each time

bertil291utn committed:
Create partial file to font and colors

bertil291utn committed:
Test with webpack
- Copy all sass files to src files
- Add css render webpack config file

bertil291utn committed:
Merge branch 'feature/design' into feature/weather-app-branch

bertil291utn committed:
Remove  saving layout page on this branch

bertil291utn committed:
Add input float in the rigth

bertil291utn committed:
Add padding to input

bertil291utn committed:
Data content design
- RWD to whole page

bertil291utn committed:
Add change degrees metric
- Add option to change C to F and vice-versa

bertil291utn committed:
Add click event listener
- Search icon

bertil291utn committed:
Display current date

bertil291utn committed:
Add design part to bundle with webpack

bertil291utn committed:
Change to module pattern

bertil291utn committed:
Disable all the async and await

bertil291utn committed:
Enable all add click eventlistener

bertil291utn committed:
Merge branch 'feature/design' into feature/weather-app-branch

bertil291utn committed:
Remove layout from git ls-files

bertil291utn committed:
Add .env file  configuration to webpack
- Add babel polyfill to async and await

bertil291utn committed:
Display all data api to card element

bertil291utn committed:
Change background image position

bertil291utn committed:
Change celsius to farenheit

bertil291utn committed:
Render new city

bertil291utn committed:
Enter a city with enter key

bertil291utn committed:
Fix linter errors

bertil291utn committed:
Update README file

bertil291utn committed:
Update README file

bertil291utn committed:
Deploy dist directory

bertil291utn committed:
Add favicon

bertil291utn committed:
Build dist folder

bertil291utn committed:
Update README file

bertil291utn committed:
Update README.md

bertil291utn committed:
Update README.md

Forison commented on src/components/search.js:
```diff
@@ -0,0 +1,65 @@
+    searchInput.addEventListener('keyup', (event) => {
+      if (event.keyCode === 13) {
+        event.preventDefault();
+        // searchButton.click();
```
#### Kindly remove this commented code as mentioned in the main review.

bertil291utn replied:
Thanks, I completely forgot

Forison reviewed (changes requested):
## Status:: Changes required❌

# 🙋‍♂️ Bertil! 

#### You have done a great job on this one, your project almost all the requirements on my checklist, but there are some issues we have to address before the project can be approved.

* [ ] It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

* [ ] Kindly address the issue with 
``` api.js:23 GET https://api.teleport.org/api/urban_areas/slug:usa/images/ 404 ```, currently the call to this API fails each time, I suspect it is the reason why the flags associated to the countries of the search results are inaccurate sometimes. Check this by inspecting your page's console in the browser.
![erro](https://user-images.githubusercontent.com/29695661/90556136-62e04480-e166-11ea-8b5f-d3e25160c71e.png)

### That is it, kindly make the changes and submit a re-review🤗.

#### Happy coding👨‍💻 .

Addo Forison

bertil291utn commented on src/controller/api.js:
```diff
@@ -0,0 +1,42 @@
+
+  const getDefaultImage = () => process.env.DEFAULT_IMAGE;
+
+  const cityBackgroundImage = async (cityName) => {
+    try {
+      const response = await fetch(
+        `https://api.teleport.org/api/urban_areas/slug:${sanitizeNames(
+          cityName
+        )}/images/`
+      );
+      const resp = await response.json();
+      return resp.photos[0].image.web;
+    } catch (error) {
+      return getDefaultImage();
+    }
+  };
```
## Get default image

If not find the city it returns the default city. It's inside try-catch block 
line 30
```javascript
} catch (error) {
      return getDefaultImage();
```

bellom replied:
Yes, you have done well.
Now, let's add a div to display to the user if the city is not found, something like `city not found`, let's try to tell the user his/her word is bogus 😄 
for some reason, I couldn't start the app locally, could that be from my end 🤔 
example; if the user should type `new yor` nothing happens on the app until you check the console.

[Optional]
Note: 
we could search for a country name, some are having the correct default image you added while some are wrong.
I am still trying to figure it out why 🤔 

Correct!
![image](https://user-images.githubusercontent.com/31897434/90573200-c346a480-e1ad-11ea-8dbd-85546143bce9.png)

Wrong!
![image](https://user-images.githubusercontent.com/31897434/90573373-3c45fc00-e1ae-11ea-9c00-888c27521407.png)

bertil291utn commented:
### Hi @Forison 
Thanks for the review
Re: API calling
- This API is used to get images according to each enter city. In the top, I added a city image
It throws an error when the city you pass the input is not found, for that case, I added a [default image](https://github.com/bertil291utn/weather-app/pull/1/files#r472450202)
![city-finded](https://user-images.githubusercontent.com/24902525/90560683-19dbc080-e165-11ea-92c5-9c3372e1cb7d.gif)
 
  - If you check returns an error when a city is not found.

###### **Note:** I have to add a loading after new searching  city, thus when you enter a new city please just wait for a second

bertil291utn committed:
Delete comment in search component

bertil291utn committed:
Merge branch 'feature/weather-app-branch' of https://github.com/bertil291utn/weather-app into feature/weather-app-branch

bertil291utn committed:
Update webpack config file
 - For production env

bellom reviewed (changes requested):
## Hi @bertil291utn   :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

bertil291utn committed:
Add message when a city is not on the API

bertil291utn committed:
Fixing trailing comma issue

mcrd25 reviewed (changes requested):
# Changes Required :mag: 

Hi @bertil291utn :wave: 

Thanks for your patience. :slightly_smiling_face: 

However, there is an issue running the app locally. See console below:
![image](https://user-images.githubusercontent.com/31193823/90725555-59ea9280-e28e-11ea-9ace-9813bc823510.png)
I believe the problem arises here when running the app locally: 
![image](https://user-images.githubusercontent.com/31193823/90725621-7b4b7e80-e28e-11ea-85e2-176e897db68c.png) Line 12 of App.js.

Please address this and ensure the app can run on other machines locally.
Let me know how I can help. :smile: 

***Please makes changes and submit another review request.
If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

bertil291utn commented:
## Hi @mcrd25 

Thanks for the review.
Relating to the [issue](https://github.com/bertil291utn/weather-app/pull/1#pullrequestreview-471287242)

I'm running my local project with Webpack dev server

![webpack-running](https://user-images.githubusercontent.com/24902525/90787381-46f5b380-e2ca-11ea-9786-45bca603dba5.gif)

###### PSD. I sent you a message through Slack, but I can't reach your response, thus I'm sending another request for review

uryelah commented on README.md:
```diff
@@ -1,39 +1,43 @@
+## Usage
+- Clone the [project 📝](https://github.com/bertil291utn/weather-app.git)
+- Install all dependencies `npm i`
+- Get all your APIs and add it to `.env` file
```
- [ ] This instruction is too vague. 
Specify the name that each API key should have a from which site to get each needed key.
The issue from the last review was related to that, make sure the information on how to run it locally is as explicit as possible.

Remember: Documentation is key!

bertil291utn replied:
Yes, I didn't know what exactly was asking my previous reviewer, and it was my error.
Thanks for the feedback

uryelah commented on src/components/search.js:
```diff
@@ -0,0 +1,79 @@
+        API.cityBackgroundImage(searchInput.value),
+      ]);
+
+      const { main, weather, wind, clouds, sys } = await returnCityData[0];
```
- [ ] [OPTIONAL] When the returnCityData doesn't return these fields, for example when it has an error related to the API key, you could add a try/catch block to show a message to the user like 'Something went wrong' instead of having the app crash.

bertil291utn replied:
Yes, I have to catch that error

uryelah reviewed (approved):
Hi **Bertil**, I'm Sarah your reviewer for this PR 👩‍🚀 

Great job with this project, I really like how you structured your code!

I'm approving your project because it worked locally after I added the API keys needed.
Make sure to read the comment below about making the instructions for running the app locally clearer.

Go forth and rock your next project!  👨‍🎤 🎸  

-------

## **Your project is approved! 🎊 🥅 💛**

**You may close this Pull Request and merge the changes**


---------

    
    
**Keep being an awesome dev and good luck with your next project!**

---------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

bertil291utn commented:
### Hi @uryelah 
Thank you for the review.
It's been one of the best💻 🔥 🚀

bertil291utn committed:
Update README file
- Remove default variable from .env file

bertil291utn committed:
Add colors to buttons
- Add new scripts to deploy and force deploy to GH pages

bertil291utn committed:
Rebuild dist file

bertil291utn committed:
Update package json file

bertil291utn committed:
Add script to deploy to gh pages
